exchange/auction.go

```diff
@@ -176,6 +175,52 @@ func (a *auction) doCache(ctx context.Context, cache prebid_cache_client.Client,
 	return errs
 }
 
+func (a *auction) cacheBid(bidToCache *pbsOrtbBid, incBannerBids bool, incVastBids bool, hasCustomCacheKey bool, toCache *[]prebid_cache_client.Cacheable, expByImp map[string]int64, defaultTTLs *config.DefaultTTLs, ttlBuffer int64, bidCategory map[string]string, hbCacheID string) (bool, bool, int, error) {
+	var chachedBid, chachedVast bool = false, false
+	var cachedSoFar int = len(*toCache)
+	var anError error = nil
+
+	if incBannerBids { //banner
+		if jsonBytes, err := json.Marshal(bidToCache.bid); err == nil {
+			if hasCustomCacheKey {
+				anError = errors.New("cannot use custom cache key for non-vast bids")
+			}
+			*toCache = append(*toCache, prebid_cache_client.Cacheable{
+				Type:       prebid_cache_client.TypeJSON,
+				Data:       jsonBytes,
+				TTLSeconds: cacheTTL(expByImp[bidToCache.bid.ImpID], bidToCache.bid.Exp, defTTL(bidToCache.bidType, defaultTTLs), ttlBuffer),
+			})
+			chachedBid = true
+		} else {
+			anError = err
+		}
+
+	}
+	if incVastBids && bidToCache.bidType == openrtb_ext.BidTypeVideo { //video
+		if jsonBytes, err := json.Marshal(makeVAST(bidToCache.bid)); err == nil {
+			_, isTopBid := a.impsToTopBids[bidToCache.bid.ImpID]
```

Comment:
If this function includes an argument that indicates the bid is already known to be a winner (which is true when iterating through impsToTopBids), then it's not necessary to make the above check when that argument is set to true.  In the long run, that should be a performance win.

Reply:
Good catch. I'm still including the check whenever the new flag is false because we'll sometimes call this function from the for statement in line 117 that traverses `a.impsToBiddersTopBids` making this check to be needed anyways